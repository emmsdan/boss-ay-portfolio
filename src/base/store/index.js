import Vue from 'vue'
import Vuex from 'vuex'

// import modules here
import infoCard from '@module/InfoCard/store'

Vue.use(Vuex)

export default new Vuex.Store({
  namespace: true,
  state: {
    darkMode: false,
    isLoggedIn: false,
    showMenu: false
  },
  mutations: {
    toggleState: (state, stateName) => {
      state[stateName] = !state[stateName]
    }
  },
  modules: {
    infoCard
  }
})
