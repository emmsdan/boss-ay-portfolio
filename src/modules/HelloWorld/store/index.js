export default {
  namespaced: true,
  state: {
    total: 20,
    title: 'Bro I dey'
  },
  mutations: {},
  actions: {},
  getters: {}
}
