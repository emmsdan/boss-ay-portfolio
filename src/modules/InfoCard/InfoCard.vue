<template>
  <section class="infocard">
    <div class="info">
      <h1> {{ getName }} </h1>
      <h2> {{ getDescription }} </h2>
      <div>
        <router-link :to="getFirstButtonUrl">
          <button> {{ getFirstButtonText }}</button>
        </router-link>
        <router-link  v-if="getSecondButtonText" :to="getSecondButtonUrl">
          <button class="stack"> {{ getSecondButtonText }} </button>
        </router-link>
      </div>
    </div>
    <div class="picture">
      <img :src="getPlaceholderImage" />
    </div>
  </section>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  name: 'InfoCardModule',
  computed: {
    ...mapGetters('infoCard', ['getPlaceholderImage', 'getSecondButtonUrl', 'getSecondButtonText', 'getFirstButtonText', 'getFirstButtonUrl', 'getDescription', 'getName'])
  },
  methods: {}
}
</script>
