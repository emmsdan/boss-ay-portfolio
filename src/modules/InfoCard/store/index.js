export default {
  namespaced: true,
  state: {
    description: 'passionate about enhancing human potential, helping people and organizations grow, and leaving a lasting positive impact on lives.',
    name: 'I am EmmsDan',
    firstButton: {
      text: 'learn more...',
      url: '/emmauel-daniel-resume.pdf'
    },
    secondButton: {
      text: 'How can I help you?',
      url: 'contact'
    },
    placeholder: '/img/logo.75891cef.png'
  },
  mutations: {
    update (state, payload) {
      state = payload
    }
  },
  actions: {},
  getters: {
    getDescription (state) {
      return state.description
    },
    getName (state) {
      return state.name
    },
    getFirstButtonText (state) {
      return state.firstButton.text
    },
    getFirstButtonUrl (state) {
      return state.firstButton.url
    },
    getSecondButtonText (state) {
      return state.secondButton.text
    },
    getSecondButtonUrl (state) {
      return state.secondButton.url
    },
    getPlaceholderImage (state) {
      return state.placeholder
    }
  }
}
