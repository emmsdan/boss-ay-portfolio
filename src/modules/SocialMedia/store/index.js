export default {
  namespaced: true,
  state: {
    description: 'passionate about enhancing human potential...',
    name: 'I am EmmsDan',
    firstButton: {
      text: 'learn more...',
      url: '/resume'
    },
    secondButton: {
      text: 'How can I help you?',
      url: 'contact'
    },
    placeholder: '/ay_logo2.png'
  },
  mutations: {
    update (state, payload) {
      state = payload
    }
  },
  actions: {},
  getters: {
    getDescription (state) {
      return state.description
    },
    getName (state) {
      return state.name
    },
    getFirstButtonText (state) {
      return state.firstButton.text
    },
    getFirstButtonUrl (state) {
      return state.firstButton.url
    },
    getSecondButtonText (state) {
      return state.secondButton.text
    },
    getSecondButtonUrl (state) {
      return state.secondButton.url
    },
    getPlaceholderImage (state) {
      return state.placeholder
    }
  }
}
