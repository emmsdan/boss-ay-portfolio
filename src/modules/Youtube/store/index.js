export default {
  namespaced: true,
  state: {},
  mutations: {
    update (state, payload) {
      state = payload
    }
  },
  actions: {},
  getters: {}
}
