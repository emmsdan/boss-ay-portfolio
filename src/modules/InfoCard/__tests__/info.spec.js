import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'

import InfoCard from '../InfoCard'
import store, { getters, noSecondButton } from './mockDate'

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(VueRouter)
describe('Plugins.vue', () => {
  let localStore, localStore2

  afterAll(() => {
    localStore = localStore2 = null
  })
  beforeEach(() => {
    localStore = new Vuex.Store({
      ...store
    })
    localStore2 = new Vuex.Store({
      ...noSecondButton
    })
  })

  it('should render component with out changes', () => {
    const wrapper = shallowMount(InfoCard, { store: localStore, localVue })
    expect(wrapper).toMatchSnapshot()
  })
  it('render heading', () => {
    const wrapper = shallowMount(InfoCard, { store: localStore, localVue })
    const h1 = wrapper.get('h1')
    expect(h1.text()).toBe(getters.getName())
  })
  it('render description text', () => {
    const wrapper = shallowMount(InfoCard, { store: localStore, localVue })
    const h2 = wrapper.get('h2')
    expect(h2.text()).toBe(getters.getDescription())
  })
  it('render placeholder image', () => {
    const wrapper = shallowMount(InfoCard, { store: localStore, localVue })
    const image = wrapper.get('img')

    expect(image).toBeTruthy()
  })
  it('render first and second button', () => {
    const wrapper = shallowMount(InfoCard, { store: localStore, localVue })
    const button1 = wrapper.findAll('button').at(0)
    const button2 = wrapper.findAll('button').at(1)
    expect(button1.text()).toBe(getters.getFirstButtonText())
    expect(button2.text()).toBe(getters.getSecondButtonText())
  })

  it('should not render second button', () => {
    const wrapper = shallowMount(InfoCard, {
      store: localStore2,
      localVue
    })
    expect(wrapper.isVisible()).toBe(true)
    const button = wrapper.find('button.stack').exists()
    expect(button).toBe(false)
  })
})
