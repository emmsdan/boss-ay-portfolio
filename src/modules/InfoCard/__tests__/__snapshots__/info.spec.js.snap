// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard.vue should render component with out changes 1`] = `
<section class="infocard">
  <div class="info">
    <h1> app name </h1>
    <h2> some-description </h2>
    <div>
      <router-link-stub to="text-content" tag="a" event="click"><button> /text-content-url</button></router-link-stub>
      <router-link-stub to="/another-text-content-url" tag="a" event="click"><button class="stack"> another-text-content </button></router-link-stub>
    </div>
  </div>
  <div class="picture"><img src="/a-lovely-image.png"></div>
</section>
`;
